SQF-VM/SQF_types.c

```diff
@@ -327,6 +327,66 @@ void array_push(PARRAY arr, VALUE val)
 	arr->data[arr->top]->val = val.val;
 	arr->top++;
 };
+
+PVALUE array_pop(PARRAY array) {
+	array->top--;
+	PVALUE val = array->data[array->top];
+	array->data[array->top] = NULL;
+
+	return val;
+}
+
+PVALUE array_popAt(PARRAY array, int index) {
+	if(index < 0 || index >= array->top) {
+		return value_create(NOTHING_TYPE(), base_int(0));
+	}
+
+	PVALUE val = array->data[index];
+
+	for(int i=index; i<array->top - 1; i++) {
+		array->data[i] = array->data[i+1];
+	}
+
+	array->top--;
+	array->data[array->top] = NULL;
+
+	return val;
+}
+
+void array_resizeSQF(PARRAY array, int newSize) {
```

Comment:
Not sure what this is supposed to mean, see L310 `void array_resize(PARRAY, unsigned int)`
please give insight about purpose

Reply:
The `array_resize` is the function that deals with mempory allocation (therefore with the actual size of the array) wheras this function mirrors the functionality of the SQF command `resize` which will "resize the content" of the array.

E.g. `[1,2,3,4] resize 2` -> `[1,2]` and `[1,2,3,4] resize 7` -> `[1,2,3,4,nil,nil,nil]`

Therefore the major differnce is that `array_resize` will change `array->size` wherasa `array_resizeSQF` will only change `array->top` (unless the array is enlarged beyond its current size of course)

Reply:
please add documentation (leading `///`) to the function